src/numeric-types.c

```diff
@@ -93,30 +84,140 @@ numeric_##type##_to_##gtype (numeric_##type num)                     \
+    numeric_type_register_static_simple (G_TYPE_UINT64, "uint64", sizeof (guint64), G_BYTE_ORDER);
+    numeric_type_register_static_simple (G_TYPE_FLOAT,  "float",  sizeof (gfloat),  G_BYTE_ORDER);
+    numeric_type_register_static_simple (G_TYPE_DOUBLE, "double", sizeof (gdouble), G_BYTE_ORDER);
+}
+
+const NumericTypeInfo *
+numeric_get_type_info (GType type)
+{
+    register_glib_numerical_types ();
+
+    gint i;
+    for (i = 0; i < TYPE_TABLE->len; i++)
+    {
+        NumericTypeInfo* ti = &g_array_index (TYPE_TABLE, NumericTypeInfo, i);
+        if (ti->type == type)
+        {
+            return ti;
+        }
+    }
+
+    g_return_val_if_reached (NULL);
+}
+
+const NumericTypeInfo *
+numeric_get_type_info_from_name (const gchar *name)
+{
+    register_glib_numerical_types ();
+
+    gint i;
+    for (i = 0; i < TYPE_TABLE->len; i++)
+    {
+        const NumericTypeInfo* ti = &g_array_index (TYPE_TABLE, NumericTypeInfo, i);
+        if (g_strcmp0 (ti->name, name) == 0)
+        {
+            return ti;
+        }
+    }
+
+    g_return_val_if_reached (NULL);
+}
+
+DEFINE_NUMERIC (int128,     G_BYTE_ORDER)
+DEFINE_NUMERIC (uint128,    G_BYTE_ORDER)
+DEFINE_NUMERIC (float80,    G_BYTE_ORDER)
+DEFINE_NUMERIC (float128,   G_BYTE_ORDER)
+DEFINE_NUMERIC (decimal32,  G_BYTE_ORDER)
+DEFINE_NUMERIC (decimal64,  G_BYTE_ORDER)
+DEFINE_NUMERIC (decimal128, G_BYTE_ORDER)
+DEFINE_NUMERIC (complex,    G_BYTE_ORDER)
+DEFINE_NUMERIC (complex80,  G_BYTE_ORDER)
+DEFINE_NUMERIC (complex128, G_BYTE_ORDER)
+
+DEFINE_NUMERIC_WITH_BYTESWAP (int_le,    G_LITTLE_ENDIAN, int,    gint,    GINT,    LE)
+DEFINE_NUMERIC_WITH_BYTESWAP (int_be,    G_BIG_ENDIAN,    int,    gint,    GINT,    BE)
+DEFINE_NUMERIC_WITH_BYTESWAP (uint_le,   G_LITTLE_ENDIAN, uint,   guint,   GUINT,   LE)
+DEFINE_NUMERIC_WITH_BYTESWAP (uint_be,   G_BIG_ENDIAN,    uint,   guint,   GUINT,   BE)
+DEFINE_NUMERIC_WITH_BYTESWAP (long_le,   G_LITTLE_ENDIAN, long,   glong,   GLONG,   LE)
+DEFINE_NUMERIC_WITH_BYTESWAP (long_be,   G_BIG_ENDIAN,    long,   glong,   GLONG,   BE)
+DEFINE_NUMERIC_WITH_BYTESWAP (ulong_le,  G_LITTLE_ENDIAN, ulong,  gulong,  GLONG,   LE)
+DEFINE_NUMERIC_WITH_BYTESWAP (ulong_be,  G_BIG_ENDIAN,    ulong,  gulong,  GLONG,   BE)
+DEFINE_NUMERIC_WITH_BYTESWAP (int64_le,  G_LITTLE_ENDIAN, int64,  gint64,  GINT64,  LE)
+DEFINE_NUMERIC_WITH_BYTESWAP (int64_be,  G_BIG_ENDIAN,    int64,  gint64,  GINT64,  BE)
+DEFINE_NUMERIC_WITH_BYTESWAP (uint64_le, G_LITTLE_ENDIAN, uint64, guint64, GUINT64, LE)
+DEFINE_NUMERIC_WITH_BYTESWAP (uint64_be, G_BIG_ENDIAN,    uint64, guint64, GUINT64, BE)
+DEFINE_NUMERIC_WITH_BYTESWAP (float_le,  G_LITTLE_ENDIAN, float,  gint32,  GINT32,  LE)
+DEFINE_NUMERIC_WITH_BYTESWAP (float_be,  G_BIG_ENDIAN,    float,  gint32,  GINT32,  BE)
+DEFINE_NUMERIC_WITH_BYTESWAP (double_le, G_LITTLE_ENDIAN, double, gint32,  GINT64,  LE)
+DEFINE_NUMERIC_WITH_BYTESWAP (double_be, G_BIG_ENDIAN,    double, gint32,  GINT64,  BE)
```

Comment:
I think we're almost good now. I will incorporate your piece of code (if license permit?) to perform generic byte swapping.

Reply:
No problem. There is a minimal version of  numpy 'dtype' in bigfile, included some macros for casting and striding. Feel free to recycle any pieces there.